core/dnsserver/onstartup.go

```diff
@@ -2,9 +2,16 @@ package dnsserver
 
 import (
 	"fmt"
+	"regexp"
 	"sort"
 )
 
+// checkDomainName() returns true if the given domain name follows RFC1035 preferred syntax.
+func checkDomainName(domainName string) bool {
+	matchDomain, _ := regexp.Compile("([A-Za-z0-9-]+\\.[A-Za-z0-9-]+)*")
```

Comment:
This will match a compliant substring in a no compliant domain name.  
Need to anchor the beginning and end of the regex.  
also pretty sure domain always ends in a dot. 

Please compile coredns with your changes and test both compliant and non-compliant domains. 
Easier / quicker to see if your code works than to iterate with reviews.

Reply:
@chrisohaver  How should I anchor beginning and ending of a domain-name?, I think ending should be anchored by a single `:` but what about beginning.

Reply:
https://www.regular-expressions.info/anchors.html

Reply:
Thanks for sharing the docs link, want to ask that, Are you sure? you want a dot `.` at the ending of Regex as ending anchor .

Reply:
Two unrelated things ...

1. The regex needs to be anchored at both ends, otherwise it could successfully match a substring of an invalid domain.  e.g. `#!@.example.com.`, would match your current regex because the substring `example.com` matches.
2. Not related to regex anchoring: the variable `domain` will always end in a dot (`.`) because it is a fully qualified domain name (normalized earlier in execution).

Reply:
I have improved(anchored beginning and ending) Regex as per your suggestion, means there must be a dot `.` at the ending of Regex for an example if `zone == an-example.com.` then `checkDomainName()` will return true or else will return false although `zone == an-example.com`. Let me know if there is still any issue